test/WebSites/InlineConstraintsWebSite/Constraints/IsbnDigitScheme10Constraint.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Routing;
+
+namespace InlineConstraintsWebSite.Constraints
+{
+    public class IsbnDigitScheme10Constraint : IRouteConstraint
+    {
+        private readonly bool _allowDashes;
+
+        public IsbnDigitScheme10Constraint(bool allowDashes)
+        {
+            _allowDashes = allowDashes;
+        }
+
+        public bool Match(
+            HttpContext httpContext,
+            IRouter route,
+            string routeKey,
+            IDictionary<string, object> values,
+            RouteDirection routeDirection)
+        {
+            var isbnRegExpression = @"^(\d{9})(\d|X)\z";
```

Comment:
Instead of `\z` use `$` to match end-of-string (it's odd to mix `^` and `\z` because `^`/`$` and `\A`/`\Z` are matching sets - never mixed).